model/item.h

```diff
@@ -94,13 +94,29 @@ namespace minsky
     /// mark item on canvas, then throw
     [[noreturn]] void throw_error(const std::string&) const;
 
+     /// @{ current rotation angle associated with this item, update bounding box if rotated
+    virtual double _rotation() const;  
+    virtual double _rotation(double);
+    ecolab::Accessor<double> rotation {
+      [this]() {return _rotation();},
+        [this](double x){return _rotation(x);}};
+    /// @} 
+    
+    
+    Item() {}
+    Item(const Item& x): m_rotation(x.m_rotation) {}
```

Comment:
You need to initialise all members and base classes to the values coming from x, not just m_rotation. It is the tediousness of that that I was trying to avoid with the ItemData trick we discussed.

Reply:
Thanks. I understand better now. I will try to implement the intermediate ItemData class again tomorrow. Unfortunately, all my attempts at it today resulted in failed compìlation.